src/plugins/share/README.mdx

```diff
@@ -0,0 +1,243 @@
+slug: /kibana-dev-docs/tutorials/share
+title: Kibana share Plugin
+summary: Introduction to Kibana "share" plugin services (sharing popup menu, URL locators, and short URLs).
+date: 2022-02-21
+tags: ['kibana', 'onboarding', 'dev', 'tutorials', 'share', 'url', 'short-url', 'locator']
+---
+
+# Share plugin
+
+The `share` plugin contains various utilities for displaying sharing context menu,
+generating deep links to other apps using *locators*, and creating short URLs.
+
+
+## Sharing context menu
+
+You can register an item into sharing context menu (which is displayed in
+Dashboard, Discover, and Visualize apps).
+
+### Example registration
+
+```ts
+import { ShareContext, ShareMenuItem } from 'src/plugins/share/public';
+
+plugins.share.register({
+  id: 'demo',
+  getShareMenuItems: (context) => [
+    {
+      panel: {
+        id: 'demo',
+        title: 'Panel title',
+        content: 'Panel content',
+      },
+      shareMenuItem: {
+        name: 'Demo list item (from share_example plugin)',
+      },
+    }
+  ],
+});
+```
+
+Now the "Demo list item" will appear under the "Share" menu in Discover and
+Dashboard applications.
+
+
+## Locators
+
+*Locators* are entities which given some *parameters* can navigate Kibana user
+to some deep link in Kibana or return back a full formatted URL.
+
+Kibana apps create locators and expose them from their plugin contracts. Then
+other plugins can use those locators to navigate deeply into Kibana applications.
+
+Locators work, both, on the server and browser. However, it is the responsibility
+of the app, which provides the locator, to register and expose the same locator
+on the server and browser.
+
+### Creating a locator for your Kibana app
+
+A Kibana app can create one or more *locators*, which given an object of *parameters*
+knows how to specify return a *Kibana location* 3-tuple. The 3-tuple consists of:
+(1) the app ID; (2) a path within the app; (3) a serializable *location state* object.
+The Kibana location 3-tuple is used to navigate the Kibana app in SPA (single page app)
+way&mdash;without reloading the page.
+
+```ts
+import type { SerializableRecord } from '@kbn/utility-types';
+import { LocatorDefinition, LocatorPublic } from 'src/plugins/share/public';
+
+export interface HelloLocatorParams extends SerializableRecord {
+  name: string;
+}
+
+export type HelloLocator = LocatorPublic<HelloLocatorParams>;
+
+export class HelloLocatorDefinition implements LocatorDefinition<HelloLocatorParams> {
+  public readonly id = 'HELLO_LOCATOR';
+
+  public readonly getLocation = async ({ name }: HelloLocatorParams) => {
+    return {
+      app: 'sampleAppId',
+      path: `/hello?name=${encodeURIComponent(name)}`,
+      state: {},
+    };
+  };
+}
+```
+
+Once your locator has been defined, you need to register it in the central locator
+registry with the `share` plugin:
+
+```ts
+const locator = plugins.share.url.locators.create(new HelloLocatorDefinition());
```

Comment:
worth mentioning why you need to register (instead of only exposing it on the contract)